weavetest/abci.go

```diff
@@ -142,6 +142,59 @@ func (w *WeaveRunner) InBlock(executeTx func(WeaveApp) error) bool {
 	return !bytes.Equal(initialHash, finalHash)
 }
 
+// ProcessAllTxs will run all included txs, split into blocksize.
+// It will Fail() if any tx returns an error, or if at any block,
+// the appHash does not change (if should change, otherwise, require it stable)
+func (w *WeaveRunner) ProcessAllTxs(txs []weave.Tx, txPerBlock int, shouldChange bool) {
+	numBlocks := numBlocks(len(txs), txPerBlock)
+	for i := 0; i < numBlocks; i++ {
+		changed := w.InBlock(func(wapp WeaveApp) error {
+			numTxs := txsInBlock(len(txs), txPerBlock, i == numBlocks-1)
+			offset := i * txPerBlock
+
+			// let's do all CheckTx first
+			for j := 0; j < numTxs; j++ {
+				tx := txs[offset+j]
+				if err := wapp.CheckTx(tx); err != nil {
+					return errors.Wrap(err, "cannot check tx")
+				}
+			}
+			// then all DeliverTx... as would be done in reality
+			for j := 0; j < numTxs; j++ {
+				tx := txs[offset+j]
+				if err := wapp.DeliverTx(tx); err != nil {
+					return errors.Wrap(err, "cannot deliver tx")
+				}
+			}
+			return nil
+		})
+		if changed != shouldChange {
+			w.t.Fatalf("unexpected change state")
+		}
+	}
+}
+
+// numBlocks returns total number of blocks for benchmarks that split b.N
+// into many smaller blocks
+func numBlocks(totalTx, txPerBlock int) int {
+	runs := totalTx / txPerBlock
+	if totalTx%txPerBlock > 0 {
+		return runs + 1
+	}
+	return runs
+}
+
+// txsInBlock will return the txPerBlock, unless this is the last block and there
+// is some remainder (so runs+1 above), where it will only return that extra, not a full block
+func txsInBlock(totalTx, txPerBlock int, lastBlock bool) int {
+	if lastBlock && (totalTx%txPerBlock > 0) {
+		return totalTx % txPerBlock
+	}
+	return txPerBlock
+}
+
+//----- mocking out ReadonlyKVStore -----//
```

Comment:
Instead of visual comment should the below part go to a separate file?

I have noticed that you like to do  visual comments in code. I think this is supposed to increase the readability but I personally never found it helpful. Isn't it better to when needed write a proper sentence explaining the **why**?

I was trying to write something smart here but I really do not know what :joy: I would like to challange [this style](https://github.com/iov-one/weave/search?q=----&unscoped_q=----) because I do not undestand it.

Reply:
We can have a larger discussion.

It is usually when there are 2-3 chunks or code in one file that are related, but inside there are groups with tight relations.

Here, I wanted to make clear where the helper code went from running InBlock-related stuff to Query related stuff... Maybe you have a better solution?

I would like something better, but I think this is somehow nicer than nothing.